Add Windows Support

joaocgreis opened the pull request:
This PR adds support to build llnode on Windows and load core files generated on other platforms. Debugging the Windows Node executable is still not working. This would be nice to have as a base to keep developing Windows support. However, if this is considered too green to land, it can serve as a straw-man for discussion and feedback is welcome.

This makes two relevant changes that are not restricted to Windows:
- Convert the llnode shortcut to javascript, to make it cross-platform. It is not possible to add a `llnode.cmd` script because npm already creates that file pointing to `llnode.sh`.
- Add an install script, instead of calling `node-gyp` directly. This is necessary to compile the module using Clang instead of CL.

Because this is not working to debug a live Windows executable, all tests fail. Still, it is possible to run `scan-test.js` with a core file generated on another platform as [described in the README](https://github.com/nodejs/llnode#useful-environment-variables). The core file must be generated using a 32bit node executable as lldb is [not ready for 64bit usage on Windows](https://lldb.llvm.org/status.html).

Ref: https://github.com/nodejs/llnode/issues/17

cjihrig commented on scripts/cleanup.js:
```diff
@@ -2,8 +2,14 @@
+const osName = require('os').type();
 
-const libExt = require('os').type() === 'Darwin' ? 'dylib' : 'so';
+let libExt = 'so';
```
Can you just make `'so'` an `else` case.

joaocgreis commented:
@cjihrig thanks, updated! Let me know if this is not what you had in mind.

joaocgreis commented:
Rebased, applied the fixup and added a `PR-URL` (previous iteration in https://github.com/joaocgreis/llnode/compare/master...joaocgreis-I5O-windows-support-0).

I understand that more reviews might be necessary before this can land, but when the time comes I won't be able to do it myself as I don't have write access in this repo.

joyeecheung commented:
It would nice if we can have some appveyor coverage for this, but I think it's ready to land. Thanks for the work on this!
